CI: Add mkosi boot tests

DaanDeMeyer opened the pull request:
Using the new mkosi Github Action, we can add some simple boot tests
for the systemd mkosi configs. This makes sure these keep working
as expected.

Also contains some extra commits fixing various bugs in the mkosi configs. (Fixes #17828)

DaanDeMeyer commented:
Opensuse has test failures (test-execute and test-seccomp) and Fedora has a dnf failure. Maybe @keszybz knows more about how to solve that one? We can remove opensuse if we don't have anyone to look into the test failures there. 

This basically copies the boot test from mkosi. With the new mkosi action, this seems like a pretty low-cost way to make sure the mkosi config images build and boot as expected. The python script is a simple pexpect script that logs into the booted container/vm and shuts it down again.

yuwata commented:
It sounds dnf is not installed. No?

DaanDeMeyer commented:
The setup-mkosi action builds dnf (and pacman) from source. It's already used in mkosi's CI to build all kinds of different distro images so I'm pretty confident the issue's not there. I don't encounter the error locally though. Maybe the dnf version built by setup-mkosi has a bug?

DaanDeMeyer commented:
Cannot reproduce the fedora failure on Fedora 33 or Arch. Updating all mkosi deps built from source to the latest versions didn't help either. One lib we don't build from source is librpm itself which comes from Ubuntu's repositories and is currently at 4.14. I'll try and build the latest librpm from source as well, maybe that fixes the issue.

bluca commented on .github/workflows/test_mkosi_boot.py:
```diff
@@ -0,0 +1,26 @@
+#!/usr/bin/env python3
+
```
\# SPDX-License-Identifier: LGPL-2.1-or-later

bluca commented:
> meson.build:904:0: ERROR: Pkg-config binary for machine MachineChoice.HOST not found. Giving up.

DaanDeMeyer commented:
I could have sworn I fixed that already. Fix incoming

bluca commented:
I imagine the current failure is due to https://github.com/systemd/systemd/pull/17846

DaanDeMeyer commented:
Nope, that one only occurs with systemd-nspawn v247 which Ubuntu Focal doesn't have. This failure is 17828, I'm trying to figure that one out but not getting very far unfortunately.

DaanDeMeyer commented:
I expected some deep and complex issue in library differences between distros causing this issue. Turns out the Fedora container simply didn't have find installed.

DaanDeMeyer commented:
(Also removed opensuse test as we removed it from mkosi's github action as well due to too much overhead in getting their package manager building from source on Focal)

DaanDeMeyer commented:
Fixes #17828

bluca commented:
Github actions are all green, nothing else is affected so no need to wait for autopkgtest, merging
